### PR TITLE
Apply CSS class to inline-editable grid cells

### LINKED ISSUE #2553: Editable Cells in a grid should indicate that they are editable
This could be something left up to the app developer, but it might be nice to have in the framework.
If built into the framework, it could be a config, either at gridmodel level or column level, that turns on a css rule on a cell when it is not being edited that helps indicate that the cell is editable.

The CSS could:
1: switch to pointer cursor on cell, to indicate that the cell would "respond in some way to a click"
2: put some special border or inset shadow on the cell to make it special.

The CSS could of course be overridden at the app level.

This feature would be especially useful in grids where only a few of the N columns are editable.  With this feature, new users don't have to guess or read the app docs to find out which columns are editable.

### PR DESCRIPTION
**Pull request by amcclain**
+ New `.xh-cell--editable` applied, but with no default styling provided by Hoist.
+ Changed previous `.xh-invalid-cell` to `.xh-cell--invalid` for consistency.
+ Fixes #2553

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Comment by amcclain** [comment]
Just pushed Toolbox update https://github.com/xh/toolbox/commit/744d1472eb628e0fe6551cf373ec6a0ac0320bd1 to `develop` to demo the use of an app-specific style to highlight editable cells - assuming we'll merge this PR or some variation on it shortly.

File: cmp/grid/columns/Column.js [hunk 1]
```diff
@@ -610,7 +610,16 @@ export class Column {
                 throw XH.exception('Column editor must be a HoistComponent or a render function');
             });
             ret.cellClassRules = {
```
Comment (cnrudd) [inline cmp/grid/columns/Column.js:612]: I think ret.cellClassRules and agOptions.cellClassRules need to specially merged so as to really be combined together.  As is now, an app's agOptions.cellClassRules will wipe out ret.cellClassRules won't it?

I have a client app with 12 instances of cellClassRules...

**Review by cnrudd — CHANGES_REQUESTED** [review]

File: cmp/grid/columns/Column.js (hunk 1 above)
Comment (amcclain) [reply cmp/grid/columns/Column.js:612]: Yes, agree 100% and that's something that I want to handle very soon via longstanding ticket #1331.

This PR doesn't change the current (suboptimal) behavior, which was introduced with the addition of the validation display to the first inline editing release.

Let me see if I can bump up that other ticket and get a clean approach into a dedicated PR for this overall Hoist release. I don't think it should hold up this merge, but agree it's a priority.

Comment (cnrudd) [reply cmp/grid/columns/Column.js:612]: ok, sounds good.

**Comment by amcclain** [comment]
Reviewed with Lee

**Pull request merged**